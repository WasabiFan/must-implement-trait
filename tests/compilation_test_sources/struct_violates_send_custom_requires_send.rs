use must_implement_trait::must_implement_trait;

fn main() {
    let _instance = MySendStruct { value: 7 };
}

trait NotSend {}

#[must_implement_trait(Send)]
struct MySendStruct {
    value: Box<dyn NotSend>
}