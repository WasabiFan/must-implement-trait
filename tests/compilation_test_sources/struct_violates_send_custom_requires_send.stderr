error[E0277]: `(dyn NotSend + 'static)` cannot be sent between threads safely
 --> $DIR/struct_violates_send_custom_requires_send.rs:9:1
  |
9 | #[must_implement_trait(Send)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `(dyn NotSend + 'static)` cannot be sent between threads safely
  |
  = help: the trait `std::marker::Send` is not implemented for `(dyn NotSend + 'static)`
  = note: required because of the requirements on the impl of `std::marker::Send` for `std::ptr::Unique<(dyn NotSend + 'static)>`
  = note: required because it appears within the type `std::boxed::Box<(dyn NotSend + 'static)>`
  = note: required because it appears within the type `MySendStruct`
  = help: see issue #48214
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
